src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -498,18 +498,19 @@ internal void ExecutePackageDownload(string name, PackageVersion package, string
             var pmExt = DynamoViewModel.Model.GetPackageManagerExtension();
             if (result == MessageBoxResult.OK)
             {
+                package.full_dependency_ids.Count == package.full_dependency_versions.Count
                 // get all of the dependency version headers
                 var dependencyVersionHeaders = package.full_dependency_ids.Select((dep, i) =>
                 {
+                    var depVersion = package.full_dependency_versions[i];
                     try
                     {
-                        var depVersion = package.full_dependency_versions[i];
                         return Model.GetPackageVersionHeader(dep._id, depVersion);
                     }
                     catch
                     {
                         MessageBoxService.Show(
-                            String.Format(Resources.MessageFailedToDownloadPackageVersion, dep._id),
+                            String.Format(Resources.MessageFailedToDownloadPackageVersion, depVersion, dep._id),
```

Comment:
```Failed to download version {0} of package with id: {1}.  Please try again and report the package if you continue to have problems.``

Reply:
Did it crash here at `dep._id` before? Was `dep` `null` in that case?

Reply:
Calling `String.Format` with the wrong number of arguments will cause an exception to be thrown. Resources.MessageFailedToDownloadPackageVersion expects 2 args and we were only passing one (dep._id)